addons/openebs/enable

```diff
@@ -33,17 +47,21 @@ HELM="$SNAP_DATA/bin/helm3 --kubeconfig=$SNAP_DATA/credentials/client.config"
 
 $HELM repo add openebs https://openebs.github.io/charts
 $HELM repo update
-$HELM -n openebs install openebs openebs/openebs \
-    --version 3.0.x \
-    --set cstor.enabled=true \
+$HELM -n $OPENEBS_NS install openebs openebs/openebs \
+    --version 3.2.x \
+    --set ndm.enabled=false \
+    --set ndmOperator.enabled=false \
     --set jiva.enabled=true \
-    --set legacy.enabled=false \
-    --set cstor.cleanup.image.tag="latest" \
-    --set cstor.csiNode.kubeletDir="$SNAP_COMMON/var/lib/kubelet/" \
+    --set jiva.storageClass.name=openebs-jiva \
+    --set jiva.storageClass.isDefaultClass=true \
```

Comment:
I would like to see this being set as an option (disabled by default). Some reasons behind this:

- OpenEBS jiva storageclass creates volumes with 3 replicas in different nodes, which silently assumes that 3 or more nodes are available in the cluster. Trying to use the storage class in smaller clusters (e.g. single-node) will fail in non-obvious ways.
- If multiple storage classes are set as default, then Kubernetes ignores them.
- Backwards-compatibility

However, I do see how it can be useful. In this case, it's best to configure this using an argument on the addon enable script (e.g. `--use-as-default-storage-class`, but feel free to use another, I'm not good at naming things).

See https://github.com/canonical/microk8s-community-addons/blob/main/addons/argocd/enable#L20-L36 for an example on how to set and use arguments in the enable script

Reply:
I agree with you. This should also be the case for the core addon for hostpath. That one also makes itself the default class. The right way should be that no addon, including core, would set themselves default as we can't control what an admin does for their own purposes.

To give an exact example, hostpath-storage is a dependency for observability, but I would like observability to use the storage class I tell it to do so. Enabling observability, enables hostpath-storage, which sets itself as default. I expect that observability by default will respect whatever I have set as default and not enable an addon I do not want it in my system.

Reply:
Jiva is not set as default class anymore. Must be explicit set to be used.